flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/OperatorCoordinatorSchedulerTest.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.flink.runtime.concurrent.ManuallyTriggeredScheduledExecutorService;
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.executiongraph.ExecutionAttemptID;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.operators.coordination.OperatorCoordinator;
+import org.apache.flink.runtime.operators.coordination.TestingOperatorCoordinator;
+import org.apache.flink.runtime.taskmanager.TaskExecutionState;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+import org.apache.flink.util.SerializedValue;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import static org.apache.flink.runtime.scheduler.SchedulerTestingUtils.takeCheckpoint;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+
+/**
+ * Tests for the integration of the {@link OperatorCoordinator} with the scheduler, to ensure the relevant
+ * actions are leading to the right method invocations on the coordinator.
+ */
+public class OperatorCoordinatorSchedulerTest extends TestLogger {
+
+	private final JobVertexID testVertexId = new JobVertexID();
+	private final OperatorID testOperatorId = new OperatorID();
+
+	private final ManuallyTriggeredScheduledExecutorService executor = new ManuallyTriggeredScheduledExecutorService();
+
+	// ------------------------------------------------------------------------
+	//  tests
+	// ------------------------------------------------------------------------
+
+	@Test
+	public void testCoordinatorStartedWhenSchedulerStarts() throws Exception {
+		final DefaultScheduler scheduler = createAndStartScheduler();
+		final TestingOperatorCoordinator coordinator = getCoordinator(scheduler);
+
+		assertTrue(coordinator.isStarted());
+	}
+
+	@Test
+	public void testCoordinatorDisposedWhenSchedulerStops() throws Exception {
+		final DefaultScheduler scheduler = createAndStartScheduler();
+		final TestingOperatorCoordinator coordinator = getCoordinator(scheduler);
+
+		scheduler.suspend(new Exception("test suspend"));
+
+		assertTrue(coordinator.isClosed());
+	}
+
+	@Test
+	public void testFailureToStartPropagatesExceptions() throws Exception {
+		final OperatorCoordinator.Provider failingCoordinatorProvider =
+			new TestingOperatorCoordinator.Provider(testOperatorId, CoordinatorThatFailsInStart::new);
+		final DefaultScheduler scheduler = createScheduler(failingCoordinatorProvider);
+
+		try {
```

Comment:
Use `(expected = Exception.class)` along with the annotation?

Reply:
I have made some bad experience with that pattern where the exception was thrown in a different place (broken mick instantiation), which made the test pass but useless. This pattern is more safe in precise, in my opinion.

Reply:
Yeah, that makes sense.